Interface.C

```diff
@@ -234,13 +237,30 @@ void preciceAdapter::Interface::configureMesh(const fvMesh& mesh, const std::str
             // Assign the (x,y,z) locations to the vertices
             // TODO: Ensure consistent order when writing/reading
             for (int i = 0; i < faceNodes.size(); i++)
+            {
                 for (unsigned int d = 0; d < dim_; ++d)
+                {
                     vertices[verticesIndex++] = faceNodes[i][d];
+                }
+                verticesMap.emplace(std::make_tuple(faceNodes[i][0], faceNodes[i][1], faceNodes[i][2]), -1);
```

Comment:
This will only work for dim_ == 2.
Can we be sure that this is always the case?

Reply:
OpenFOAM always works in 3D. We also have a pseudo-2D mode, where we still have full 3D coordinates and two layers in the z-direction (to define complete cells), but only operate on the one layer for face nodes. We assume the z-axis to be the "dead" axis and assume always z=0.

We still give 2D data to preCICE in that case.